File: cases/taint/javascript/infeasible-branch-negative/InfeasibleBranchNegative.js
function dfb_source() { // DFB-SOURCE: infeasible-branch-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: infeasible-branch-sink

function run() {
  let value = "clean";
  if (false) {
    value = dfb_source(); // DFB-WITNESS: infeasible-tainted-branch
  }
  dfb_sink(value);
}

File: cases/taint/csharp/infeasible-branch-negative/InfeasibleBranchNegative.cs
namespace DataFlowBench;

static class InfeasibleBranchNegative
{
    static int dfb_source() // DFB-SOURCE: infeasible-branch-input
    {
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: infeasible-branch-sink

    static void Run()
    {
        int value = 0;
        if (false)
        {
            value = dfb_source(); // DFB-WITNESS: infeasible-tainted-branch
        }
        dfb_sink(value);
    }
}
